### ISSUE TITLE
Update current course changes the course name

### ISSUE THREAD

**Issue by valleyofdawn** [IsraelHikingMap/Site#638]
Bug

When sharing a course, and checking the "update current share" I expected the change the  name of the shared course to the current one as well, instead it left it as "מסלול 1".

**Comment by HarelM** [comment]
I don't understand the steps to reproduce, can you please elaborate (the template is there for a reason...)?

**Comment by valleyofdawn** [comment]
Something between a bug and a feature

## What I expect to happen
When I update a share I expect it to have the same name as the last time I shared it

## What really happened
It changes its name to "section 1"

## What I did that caused the issue - step by step
Created a route. Shared it and named it "Awesome route"
Changed the route a little
Shared again.
Pressed "Udate current share"
Got the name "section 1" instead of "Awesome route" in the title.

## My environment
Chrome on win 10

**Comment by HarelM** [comment]
"section 1" is the name of the route that you tried to share and update and not the share title, right?
In any case, I think I saw it when developing this feature and forgot to fix it, so this is bug as far as I'm concerned, thanks for reporting!

**Comment by zstadler** [comment]
Please define the expected behavior in the following scenario:

1. A route named "Route 1" is shared with the share title "Share 1".
2. Later, "Share 1" is opened.
3. The route is modified and the route name is changed to "Route 2".
4. The "Share" button is clicked and the share window is shown
   - Is the "Update current share" checkbox clear by default?
   - What is the Title of the share: "Share 1", "Route 2", or something else?
5. The user clicks on the "Update current share" checkbox.
   - Does the Title of the share change?
   - What is the Title of the share: "Share 1", "Route 2", or something else?
6. The user clicks again on the "Update current share" checkbox.
   - Does the Title of the share change?
   - What is the Title of the share: "Share 1", "Route 2", or something else?

Note that the "Generate A URL To Share With Your Friends!" button is never clicked in the above scenario.

**Comment by HarelM** [comment]
I have implemented the following, which I believe is the required behavior:
 * When sharing a route for the first time the initial title and description will be from the selected route (this is not new).
Term: "re-sharing" - either update or create a new share based on a previously share.
re-sharing is possible only in the following scenarios: 
 * When opening a share that is yours using a link
 * After sharing while using the same browser session (no refresh).
Bottom line:
When re-sharing: the initial title and description will be those of the original share regardless of the state of the checkbox.

**Comment by zstadler** [comment]
@HarelM,

Does it mean that
1. The "current share" is set only by:
  A. Opening a share link by the share owner
  B. Crating a new share
2. The "current share" is cleared by a refresh
3. The "current share" is not affected by
  A. Adding contents of other shares (the "+" button in the shares list)
  B. Adding routes and/or POIs from files
  C. Adding or deleting user layers

**Comment by HarelM** [comment]
Yes, assuming I didn't introduce new bugs :-)
